sbi/analysis/plot.py

```diff
@@ -63,21 +301,270 @@ def pairplot(
+                            Z = probs2contours(Z, opts["contour_offdiag"]["levels"])
+                        else:
+                            Z = (Z - Z.min()) / (Z.max() - Z.min())
+                        h = plt.contour(
+                            X,
+                            Y,
+                            Z,
+                            origin="lower",
+                            extent=[
+                                limits[col][0],
+                                limits[col][1],
+                                limits[row][0],
+                                limits[row][1],
+                            ],
+                            colors=opts["samples_colors"][n],
+                            levels=opts["contour_offdiag"]["levels"],
+                        )
+                    else:
+                        pass
+                elif opts["upper"][n] == "scatter":
+                    h = plt.scatter(
+                        v[:, col],
+                        v[:, row],
+                        color=opts["samples_colors"][n],
+                        **opts["scatter_offdiag"],
+                    )
+                elif opts["upper"][n] == "plot":
+                    h = plt.plot(
+                        v[:, col],
+                        v[:, row],
+                        color=opts["samples_colors"][n],
+                        **opts["plot_offdiag"],
+                    )
+                else:
+                    pass
+
+    return _arrange_plots(
+        diag_func, upper_func, dim, limits, points, opts, fig=fig, axes=axes
+    )
+
+
+def marginal_plot(
+    samples: Union[
+        List[np.ndarray], List[torch.Tensor], np.ndarray, torch.Tensor
+    ] = None,
+    points: Optional[
+        Union[List[np.ndarray], List[torch.Tensor], np.ndarray, torch.Tensor]
+    ] = None,
+    limits: Optional[Union[List, torch.Tensor]] = None,
+    subset: List[int] = None,
+    diag: Optional[str] = "hist",
+    figsize: Tuple = (10, 10),
+    labels: Optional[List[str]] = None,
+    ticks: Union[List, torch.Tensor] = [],
+    points_colors: List[str] = plt.rcParams["axes.prop_cycle"].by_key()["color"],
+    fig=None,
+    axes=None,
+    **kwargs,
+):
+    """
+    Plot samples in a row showing 1D marginals of selected dimensions.
+
+    Each of the plots can be interpreted as a 1D-marginal of the distribution
+    that the samples were drawn from.
+
+    Args:
+        samples: Samples used to build the histogram.
+        points: List of additional points to scatter.
+        limits: Array containing the plot xlim for each parameter dimension. If None,
+            just use the min and max of the passed samples
+        subset: List containing the dimensions to plot. E.g. subset=[1,3] will plot
+            plot only the 1st and 3rd dimension but will discard the 0th and 2nd (and,
+            if they exist, the 4th, 5th and so on).
+        diag: Plotting style for diagonal, {hist, cond, None}.
```

Comment:
maybe keep the arg name for consistency here like you did, but change the doc string accordingly? sth like "Plotting style for 1D marginals {hist, kde, cond, None}"